Aplib.Core/Desire/Goal.cs

```diff
@@ -0,0 +1,79 @@
+namespace Aplib.Core.Desire
+{
+    public class Goal
+    {
+        /// <summary>
+        /// The goal is considered to be completed, when the distance of the
+        /// <see cref="CurrentHeuristics"/> is below this value.
+        /// </summary>
+        private const float Epsilon = 0.005f;
```

Comment:
Not sure if I missed a memo, but didn't we do camelCase for variables? So wouldn't this be epsilon? There are some more variables that start with an uppercase letter so ill just comment this one.

Reply:
According to the [C# Identifier Name](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names#naming-conventions) document, all constant values should be named using PascalCase.

> Use PascalCase for constant names, both fields and local constants.

Reply:
Summary of the most important keywords (I'll put this on the Wiki).

## Access modifiers.
Access modifiers in order of priority when combining them ([Microsoft C# Docs: Code-style rule options](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/code-style-rule-options?view=vs-2017)):
```cs
public
private
protected
internal
file
static
extern
new
virtual
abstract
sealed
override
readonly
unsafe
required
volatile
async
```
**Examples:**
```cs
private static readonly bool s_foo;
protected new virtual int bar;
public abstract class Baz;
```

## Identifier names.
Naming conventions for identifiers ([Microsoft C# Docs: Identifier names](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names), [Microsoft C# Docs: protected](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/protected)):

```cs
namespace PascalCase { ... } // Namespaces.
<TPascalCase> // Generic types. Prefix 'T'.
interface IPascalCase { ... } // Interfaces. Prefix 'I'.
class PascalCase { ... } // Classes.
public class PascalCaseAttribute: Attribute { ... } // Attributes. Suffix 'Attribute'.
... PascalCase( ... camelCase ) { ... } // Methods and method arguments.
... PascalCase { get; set; } // Properties.
const PascalCase; // Constants.
... camelCase; // Local variables.
public static s_PascalCase; // Public static fields. Prefix 's_'.
protected static s_camelCase; // Protected static fields. Prefix 's_'.
private static s_camelCase; // Private static fields. Prefix 's_'.
public PascalCase; // Public non-static fields.
protected camelCase; // Protected non-static fields.
private _camelCase; // Private non-static fields. Prefix '_'.
```
**Additional naming conventions:**
- Use meaningful and descriptive names for variables, methods, and classes.
- Prefer clarity over brevity.
- Enum types use a singular noun for nonflags, and a plural noun for flags.
- Identifiers shouldn't contain two consecutive underscore (_) characters. Those names are reserved for compiler-generated identifiers.
- Avoid using abbreviations or acronyms in names, except for widely known and accepted abbreviations.
- Use meaningful and descriptive namespaces that follow the reverse domain name notation.
- Choose assembly names that represent the primary purpose of the assembly.
- Avoid using single-letter names, except for simple loop counters.

## Order of members (we have yet to decide on this, but for example).
The ordering of class members is as follows ([StyleCop Docs](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1201.md)).

**Within a class, struct or interface:**
- Constant Fields
- Fields
- Constructors
- Finalizers (Destructors)
- Delegates
- Events
- Enums
- Interfaces (interface implementations)
- Properties
- Indexers
- Methods
- Structs
- Classes

**Within each of these groups, order by access:**
- Public.
- Internal.
- Protected internal.
- Protected.
- Private.

**Within each of the access groups, order by static, then nonstatic:**
- Static
- Nonstatic

**Within each of the static/nonstatic groups of fields, order by readonly, then nonreadonly:**
- Readonly
- Nonreadonly

Reply:
@SSelinn does this resolve this conversation?